src/azure-cli-core/azure/cli/core/tests/test_generic_update.py

```diff
@@ -39,69 +70,71 @@ def my_set(**kwargs): #pylint:disable=unused-argument
+        self.assertEqual(my_obj.my_list_of_camel_dicts[1]['myKey'], 'new_value', 'index into list of dictionaries by camel-case key')
 
-        app.execute('gencommand --set list[2][1].d=h'.split())
-        self.assertEqual(my_obj['list'][2][1]['d'], 'h', 'set nested dict element')
+        app.execute('update-obj --set myListOfSnakeDicts[my_key=value2].my_key=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_snake_dicts[1]['my_key'], 'new_value', 'index into list of dictionaries by snake-case key')
 
-        app.execute('gencommand --set list[0]={} list[0].foo=bar'.split())
-        self.assertEqual(my_obj['list'][0]['foo'], 'bar', 'replace nested scalar with new dict')
+        app.execute('update-obj --set myListOfObjects[myString=1.2.3.4].myString=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_objects[1].my_string, 'new_value', 'index into list of objects by key')
 
-        app.execute('gencommand --set list[1]=[] --add list[1] key1=value1'
-                    ' --set list[1][0].key2=value2'.split())
-        self.assertEqual(my_obj['list'][1][0]['key1'], 'value1',
-                         'replace nested scalar with new list with one value')
-        self.assertEqual(my_obj['list'][1][0]['key2'], 'value2',
-                         'add a second value to the new list')
+        # Test setting on elements nested within lists
+        app.execute('update-obj --set myList[1][1]=newValue'.split())
+        self.assertEqual(my_obj.my_list[1][1], 'newValue', 'set nested list element')
 
-        app.execute('gencommand --set list[2][d=f].d=g'.split())
-        self.assertEqual(my_obj['list'][2][2]['d'], 'g',
-                         'index dictionary by unique key=value')
+        app.execute('update-obj --set myList[2].myKey=newValue'.split())
+        self.assertEqual(my_obj.my_list[2]['myKey'], 'newValue', 'set nested dict element')
 
-        app.execute('gencommand --add list i=j k=l'.split())
-        self.assertEqual(my_obj['list'][-1]['k'], 'l',
-                         'add multiple values to a list at once (verify last element)')
-        self.assertEqual(my_obj['list'][-1]['i'], 'j',
-                         'add multiple values to a list at once (verify first element)')
+        app.execute('update-obj --set myList[3].myInt=50'.split())
+        self.assertEqual(my_obj.my_list[3].my_int, 50, 'set nested object element')
 
-        app.execute('gencommand --add list[-2] prop2=val2'.split())
-        self.assertEqual(my_obj['list'][-2][-1]['prop2'], 'val2', 'add to list')
-        app.execute('gencommand --add list[-2] prop3=val3'.split())
-        self.assertEqual(my_obj['list'][-2][-1]['prop3'], 'val3',
-                         'add to list again, should make seperate list elements')
+        # Test overwriting and removing values
+        app.execute('update-obj --set myProp={} myProp.foo=bar'.split())
```

Comment:
>update-obj --set myProp={} myProp.foo=bar [](start = 21, length = 41)

it would be good to have somehting like:
--set myProp=None --add myProp 'str1' 'str2' --remove myProp[0] -> verify myProp is list of length 1 with str2 as first element
this tests the setting to an null object, the order of operations and mixing the operations